components/tools/OmeroWeb/omeroweb/webadmin/views.py

```diff
@@ -588,17 +588,15 @@ def manage_experimenter(request, action, eid=None, conn=None, **kwargs):
 
                 # Update 'AdminPrivilege' config roles for user
                 privileges = conn.get_privileges_from_form(form)
-                if privileges is None:
-                    privileges = []
                 # Only process privileges that we have permission to set
                 to_add = []
                 to_remove = []
-                for p in conn.getCurrentAdminPrivileges():
-                    if p in privileges:
-                        to_add.append(p)
-                    else:
-                        to_remove.append(p)
-
+                if privileges is not None:
+                    for p in conn.getCurrentAdminPrivileges():
```

Comment:
How come it's relative to the current admin privileges, not those of the target user `experimenter`? Maybe I am just not understanding the gateway.

Reply:
I think this is about privilege escalation. I can't give someone a permission that I don't have myself.

Reply:
With the code as it is now, I can only add or remove a privilege from someone if I have the privilege myself. Should you be able to remove a privilege from someone if you don't have the privilege yourself? cc @pwalczysko

Reply:
>Should you be able to remove a privilege from someone if you don't have the privilege yourself? cc @pwalczysko

IMHO the answer is no.  But the implementation as it stands now is different, which we simply took in our stride back in summer 2017 when this was first rolled out. Also, note the two problems specified in https://github.com/openmicroscopy/openmicroscopy/pull/5780#issuecomment-396896942 -> we are half allowing this, half failing in the process when this is acted upon in the UI

Reply:
I still don't understand the logic. For instance, why do we add a privilege as `toRemove` just because we don't have it?

Reply:
We don't add a privilege to remove unless we have it AND it's not chosen in the form for the user we're editing.
We go through all the privileges the I have ```for p in conn.getCurrentAdminPrivileges():``` since those are the only one's I can edit for another user.
For each privilege, if it was selected in the form, we add it, otherwise we remove it.

Reply:
Ah, thank you, that makes sense.